File: codeforces/problemset/C. Make Equal Again.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int t;
    scanf("%d",&t);
    while(t--){
        int n;
    scanf("%d",&n);
    vector<int>a(n);
    for(auto &it:a)
        scanf("%d",&it);
    
    int cnt1 =0 ,cnt2 = 0;

    int i = 0;
    while(a[i] == a[0]){
        cnt1++,i++;
    }

    i = a.size() - 1;
    while(a[i] == a.back()){
        cnt2++,i--;
    }

    if(a.front() == a.back()){
        if(cnt1 + cnt2 >= n){
            puts("0");
        }else{
            printf("%d\n",n - cnt1 - cnt2);
        }
    }else{
        printf("%d\n",max(cnt1,cnt2));
    }

    }

    return 0;
}

File: codeforces/problemset/C. Make Equal Again.java
import java.util.Scanner;
import java.util.Vector;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int t = scanner.nextInt();
        while (t-- > 0) {
            int n = scanner.nextInt();
            Vector<Integer> a = new Vector<>(n);
            for (int i = 0; i < n; i++) {
                a.add(scanner.nextInt());
            }
            
            int cnt1 = 0, cnt2 = 0;
            
            int i = 0;
            while (i < n && a.get(i).equals(a.get(0))) {
                cnt1++;
                i++;
            }
            
            i = n - 1;
            while (i >= 0 && a.get(i).equals(a.get(n - 1))) {
                cnt2++;
                i--;
            }
            
            if (a.firstElement().equals(a.lastElement())) {
                if (cnt1 + cnt2 >= n) {
                    System.out.println("0");
                } else {
                    System.out.println(n - cnt1 - cnt2);
                }
            } else {
                System.out.println(Math.max(cnt1, cnt2));
            }
        }
        
        scanner.close();
    }
}
